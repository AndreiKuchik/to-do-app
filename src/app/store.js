import { configureStore } from '@reduxjs/toolkit'

import toDoListReducer from '../features/toDoListSlice'

export default configureStore({
  reducer: {
    toDoList: toDoListReducer
  },
})
