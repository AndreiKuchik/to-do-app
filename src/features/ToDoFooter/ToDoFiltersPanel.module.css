.filterButton {
    margin-left: auto;
    cursor: pointer;
    padding: 0.2rem;
    border: 1px #373537;
    background-color: white;
    color: #a3a5b2;
    border-radius: 12px;
    margin-right: 1rem;
  }
  