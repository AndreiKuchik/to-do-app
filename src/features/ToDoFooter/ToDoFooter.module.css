.footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.5rem;
}

.footer span {
    padding: 0.2rem;
    font-size: small;
    background-color: white;
    color: #a3a5b2;
}

.completeAllButton {
    cursor: pointer;
    padding: 0.2rem;
    border: 1px;
    background-color: white;
    color: #a3a5b2;
    border-radius: 1rem;
    margin-right: 1rem;
  }
  
