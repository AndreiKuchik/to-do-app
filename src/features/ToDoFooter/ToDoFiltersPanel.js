import React from "react";
import styles from "./ToDoFiltersPanel.module.css";
import { changeFilter } from "../toDoListSlice";
import { Filters } from "../../app/filtersEnum";
import { useDispatch } from "react-redux";

export const ToDoFiltersPanel = () => {
  const dispatch = useDispatch();

  return (
    <div>
      <button
        type="button"
        className={styles.filterButton}
        onClick={() => dispatch(changeFilter(Filters.All))}
      >
        All
      </button>
      <button
        type="button"
        className={styles.filterButton}
        onClick={() => dispatch(changeFilter(Filters.Active))}
      >
        Active
      </button>
      <button
        type="button"
        className={styles.filterButton}
        onClick={() => dispatch(changeFilter(Filters.Completed))}
      >
        Completed
      </button>
    </div>
  );
};
