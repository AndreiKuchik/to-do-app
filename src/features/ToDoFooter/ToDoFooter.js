import React from 'react';
import PropTypes from 'prop-types';
import { useAppDispatch, useAppSelector } from '../../core/hooks';
import ToDoFiltersPanel from './ToDoFilterPanel/ToDoFiltersPanel';
import { deleteAllCompleted } from '../toDosSlice';
import { selectToDoListAmount } from '../toDosSelectors';
import styles from './ToDoFooter.module.css';

const ToDoFooter = () => {
  const dispatch = useAppDispatch();
  const toDoListLenght = useAppSelector(selectToDoListAmount);

  return (
    <div className={styles.footer}>
      <span role ='to-do-item-amount'>{toDoListLenght} Item left</span>
      <ToDoFiltersPanel />
      <button
        className={styles.completeAllButton}
        type="button"
        role="clear-completed-button"
        onClick={() => dispatch(deleteAllCompleted())}
      >
        Clear Completed
      </button>
    </div>
  );
};

ToDoFooter.propTypes = {
  toDosAmount: PropTypes.number,
};

export default ToDoFooter;
