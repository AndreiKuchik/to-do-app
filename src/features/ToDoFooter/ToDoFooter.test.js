import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ToDoFooter from './ToDoFooter';
import store from '../../app/store';

describe('Testing ToDoFooter component', () => {
  it('Matches DOM Snapshot', () => {
    const domTree = renderer
      .create(
        <Provider store={store}>
          <ToDoFooter />
        </Provider>,
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
