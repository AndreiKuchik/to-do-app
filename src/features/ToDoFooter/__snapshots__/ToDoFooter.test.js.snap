// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ToDoFooter component Matches DOM Snapshot 1`] = `
<div>
  <div
    class="footer"
  >
    <span
      role="to-do-item-amount"
    >
      4
       Item left
    </span>
    <div>
      <button
        class="filterButton"
        type="button"
      >
        All
      </button>
      <button
        class="filterButton"
        role="active-filter-button"
        type="button"
      >
        Active
      </button>
      <button
        class="filterButton"
        role="completed-filter-button"
        type="button"
      >
        Completed
      </button>
    </div>
    <button
      class="completeAllButton"
      role="clear-completed-button"
      type="button"
    >
      Clear Completed
    </button>
  </div>
</div>
`;
